lib/src/repo.rs

```diff
@@ -1353,6 +1353,17 @@ impl MutableRepo {
         Ok(())
     }
 
+    pub fn rename_workspace(
+        &mut self,
+        old_workspace_id: &WorkspaceId,
+        new_workspace_id: WorkspaceId,
+        wc_commit_id: CommitId,
```

Comment:
I think `wc_commit_id` argument is redundant. We just need to `wc_commit_ids.remove()`/`remove_entry()` and `.insert()` the removed entry.

Reply:
Good idea. I added a `RenameWorkspaceError` type to handle the case where wc_commit_ids does not contain the workspace id. Let me know if there's a better way to do this.

Reply:
`.rename_remote()` ignores error, but I like the explicit error. Maybe we should also check `new_workspace_id` does not exist (which will replace precondition in CLI.)

Reply:
Done.